### PR TITLE
feat(about): add About Page

### PR DESCRIPTION
**Pull request by gep13**

### CODE REVIEW

**Comment by gep13** [comment]
@Odonno what are your thoughts about the suggested About Page?

**Comment by gep13** [comment]
I have added you as a collaborator on the fork, so if you wanted to make changes before pulling in, you can certainly do that, or if you want me to make some changes, I would be happy to make them.

Couple things about the implementation:

- After navigating to the About Page, clicking the back button exits the application.  This suggests that the MainPage entry is being removed from the BackStack, but I couldn't see any "obvious" code that was doing this.  Any ideas?
- I would have liked to pull more information from the `Windows.ApplicationModel` for displaying in the About Page, but I couldn't seem to find the information.  Do you know if things like Description and Title can be easily pulled out?  If not, I can look to pull this information out manually by loading up the XML at runtime
- With regard to the History page, so you have information about what was contained in each release, so that this can be added?
- What email address would you like to be used for the feedback/email link, or do you not want this option?
- Should the About Page be navigatable from each page, i.e. Main, Room, etc?  Currently, it is only from Main

Please let me know if you have any additional comments.

**Comment by Odonno** [comment]
Nice !

Just one thing, I hate Pivot, Pivot is for WP8.0. Could you go for a Hub instead ?
And, about `AboutPage.xaml.cs`, I think it lacks navigation methods, it could break the navigation system, check other page code-behind.

**Comment by gep13** [comment]
Also, when trying to build, I am getting the following:

image

Are you also getting this?

**Comment by Odonno** [comment]
About the build error, I have the one also, I'll check soon. But it does not affect the WP version, right ?

**Comment by gep13** [comment]
Yes, you are correct, the build error is not currently affecting the Windows Phone application.

**Comment by Odonno** [comment]
And why not a Collaborator page, a list of GitHub collaborators.

**Comment by gep13** [comment]
>And why not a Collaborator page, a list of GitHub collaborators.

That seems like a good idea :+1:

**Comment by gep13** [comment]
@Odonno ok, I have implemented the suggestions that you have made above, can you take another look and let me know what you think?

With regard to the Hub, it is not immediately obvious that there is another section.  Do I have to do anything to make this more obvious?

Can you add the details about the releases to the history page?

**Comment by Odonno** [comment]
Please, don't do --amend, I will be able to see the changes as delta so.

**Comment by gep13** [comment]
@Odonno ah, sorry, I thought you might prefer to see this feature as a single commit, rather than multiple smaller commits.

**Comment by Odonno** [comment]
@gep13 GitHub is very great ! I can do a `git diff` of the entire PR.

**Comment by gep13** [comment]
@Odonno ok, I think I have caught everything that you suggested should be changed.

File: Gitter/Gitter/Gitter.WindowsPhone/Views/AboutPage.xaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                                       Style="{ThemeResource TitleTextBlockStyle}"
+                                       TextWrapping="Wrap" />
+                                <HyperlinkButton Grid.Row="2"
+                                             Content="Having Problems? Click to raise an issue"
+                                             Margin="0,12,0,0"
+                                             HorizontalAlignment="Left"
+                                             FontSize="17.333"
+                                             FontWeight="Bold"
+                                             FontFamily="Segoe WP"
+                                             NavigateUri="https://github.com/Odonno/Modern-Gitter/issues/new"/>
```
Comment (Odonno) [inline Gitter/Gitter/Gitter.WindowsPhone/Views/AboutPage.xaml]: https://github.com/Odonno/Modern-Gitter/issues/ will be better, some people are not logged in by default...

File: Gitter/Gitter/Gitter.WindowsPhone/Views/AboutPage.xaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                                    <TextBlock Text="{Binding ApplicationVersion}"
+                                           Style="{ThemeResource TitleTextBlockStyle}"/>
+                                </StackPanel>
+
+                                <TextBlock Grid.Row="1"
+                                       Text="{Binding ApplicationDescription}"
+                                       Style="{ThemeResource TitleTextBlockStyle}"
+                                       TextWrapping="Wrap" />
+                                <HyperlinkButton Grid.Row="2"
+                                             Content="Having Problems? Click to raise an issue"
```
Comment (Odonno) [inline Gitter/Gitter/Gitter.WindowsPhone/Views/AboutPage.xaml]: Use strings localization with x:Uid (check on other pages and `Strings` folder to do the translation).
In french: "Un bug ? Une suggestion ?"

File: Gitter/Gitter/Gitter.Shared/ViewModel/Concrete/AboutViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+{
+    public class AboutViewModel : ViewModelBase, IAboutViewModel
+    {
+        #region Properties
+
+        public string ApplicationDescription
+        {
+            get
+            {
+                return "A Gitter client application for Windows Phone 8";
```
Comment (Odonno) [inline Gitter/Gitter/Gitter.Shared/ViewModel/Concrete/AboutViewModel.cs]: Same for this, localize the string ! Do not put it on VM.

File: Gitter/Gitter/Gitter.Shared/ViewModel/Concrete/AboutViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                return
+                    $"{Windows.ApplicationModel.Package.Current.Id.Version.Major}.{Windows.ApplicationModel.Package.Current.Id.Version.Minor}.{Windows.ApplicationModel.Package.Current.Id.Version.Build}";
+            }
+        }
+
+        public string ApplicationPublisher
+        {
+            get
+            {
+                return "David BOTTIAU";
```
Comment (Odonno) [inline Gitter/Gitter/Gitter.Shared/ViewModel/Concrete/AboutViewModel.cs]: Remove it from the VM

File: Gitter/Gitter/Gitter.Shared/ViewModel/Concrete/AboutViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            }
+        }
+
+        public string ApplicationTitle
+        {
+            get
+            {
+                return "Modern Gitter";
```
Comment (Odonno) [inline Gitter/Gitter/Gitter.Shared/ViewModel/Concrete/AboutViewModel.cs]: Again, remove it from the VM

File: Gitter/Gitter/Gitter.Shared/ViewModel/Concrete/AboutViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                return "A Gitter client application for Windows Phone 8";
+            }
+        }
+
+        public string ApplicationVersion
+        {
+            get
+            {
+                return
+                    $"{Windows.ApplicationModel.Package.Current.Id.Version.Major}.{Windows.ApplicationModel.Package.Current.Id.Version.Minor}.{Windows.ApplicationModel.Package.Current.Id.Version.Build}";
```
Comment (Odonno) [inline Gitter/Gitter/Gitter.Shared/ViewModel/Concrete/AboutViewModel.cs]: I assume you can make a using ? `using Windows.ApplicationModel;`

File: Gitter/Gitter/Gitter.WindowsPhone/Views/MainPage.xaml [hunk 7]
```diff
@@ -105,20 +105,22 @@
...
+                <AppBarButton x:Uid="/UIResources/refresh" Label="refresh"
                               Command="{Binding Path=RefreshCommand}"
                               Icon="Refresh" />
             </CommandBar.PrimaryCommands>
 
             <CommandBar.SecondaryCommands>
-                <AppBarButton x:Uid="/UIResources/chatWithUs" Label="chat with us" 
+                <AppBarButton x:Uid="/UIResources/chatWithUs" Label="chat with us"
                               Command="{Binding Path=ChatWithUsCommand}" />
+                <AppBarButton x:Uid="/UIResources/about" Label="about"
```
Comment (Odonno) [inline Gitter/Gitter/Gitter.WindowsPhone/Views/MainPage.xaml]: Same here, `about` is `à propos` in French.

File: Gitter/Gitter/Gitter.WindowsPhone/Views/AboutPage.xaml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    <!--LayoutRoot is the root grid where all page content is placed-->
+    <Grid x:Name="LayoutRoot" Background="Transparent">
+        <Grid.RowDefinitions>
+            <RowDefinition Height="Auto"/>
+            <RowDefinition Height="*"/>
+        </Grid.RowDefinitions>
+
+        <!--ContentPanel - place additional content here-->
+        <Grid x:Name="ContentPanel" Grid.Row="1" Margin="12,0,12,0">
+            <Hub Header="About">
```
Comment (Odonno) [inline Gitter/Gitter/Gitter.WindowsPhone/Views/AboutPage.xaml]: * Remove Hub Header (useless)
* Improve font-size of section header (check properties of MainPage header)
* Every header should be lowercase (information no Information)

**Comment by bobmulder** [comment]
image

Thanks guys, I really like the annoying sound of my notifications of new e-mails :P

File: Gitter/Gitter/Gitter.WindowsPhone/Views/AboutPage.xaml [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+                                </StackPanel>
+                            </Grid>
+                        </ScrollViewer>
+                    </DataTemplate>
+                </HubSection>
+                <HubSection Header="History">
+                    <DataTemplate>
+                        <ScrollViewer>
+                            <StackPanel>
+                                <TextBlock Style="{ThemeResource SubheaderTextBlockStyle}">Version 1.2.7</TextBlock>
```
Comment (Odonno) [inline Gitter/Gitter/Gitter.WindowsPhone/Views/AboutPage.xaml]: A better history log : 

* #0.3 (soon)
    * fix bug on realtime messages
    * fix bug on toast notification
    * parse effectively all the HTML sent (title, links, image, quote, formatted code)
    * search rooms by name
    * mention (toast) notifications
    * about page
* #0.2
    * copy/update/remove message (holding on a message)
    * toast notifications (unread messages)
    * toast notifications (realtime messages)
    * in-app notification as badge (unread count)
    * 'respond to' a chat user (tap on a chat avatar)
    * microphone enabled
* #0.1
    * main page (v1)
    * room page (v1)
    * push a message
    * add a link to this chat room (chat with us)

**Comment by Odonno** [comment]
@bobmulder That's all for me ! :)

**Comment by gep13** [comment]
@Odonno with regard to the version numbers that you mention above, how do they relate to the version number in the package manifest file?  From what I can see, Modern Gitter is currently at 1.2.8, how does that tie up to 0.1, 0.2, etc?  Thanks!

**Comment by gep13** [comment]
@Odonno I have implemented some more of your suggestions, but not yet all of them.  I need to sign off for the night, but I will try to finish this up tomorrow.  Thanks for all your help! :smile_cat:

**Comment by Odonno** [comment]
> @Odonno with regard to the version numbers that you mention above, how do they relate to the version number in the package manifest file? From what I can see, Modern Gitter is currently at 1.2.8, how does that tie up to 0.1, 0.2, etc? Thanks!

Holy crap ! You're right. I've started from 0.x base but it goes to 1.x instead... Please consider my version number like this #1.1, #1.2 and #1.3.

**Comment by gep13** [comment]
When I remove the Header from the Hub, as you suggested, I end up with this:

image

And even though I try to apply the same FontSize etc from the MainPage header, they don't seem to have any effect.  Can you make any suggestions as to how to correct this?  Although, is there a way to make it clearer that there is another Hub Section?

**Comment by Odonno** [comment]
Ok, to complete the page, you need : 

* to remove the two useless grid
* to add the `Margin="0 24 0 0"` property to the Hub
* instead of `FontSize`, set `HeaderTemplate="{StaticResource HeaderTemplate}"` to each HubSection

**Comment by gep13** [comment]
@odonno ok, this time, I think I have got everything.

I have re-worded the commits, and rebased all the code review into a single commit.

Let me know if there is anything left outstanding.

**Comment by Odonno** [comment]
Rename your last commit and it's ok !

**Comment by gep13** [comment]
Ok, assuming I have understood the conventions correctly regarding the commit messages, I think this is good to go :smile:

**Comment by gep13** [comment]
Sweet! Thanks for pulling this in, we got there in the end :smile_cat:

**Pull request merged**